src/toil/jobStores/abstractJobStore.py

```diff
@@ -495,8 +495,26 @@ def getConnectedJobs(jobGraph):
             # Delete the job
             self.delete(jobGraph.jobStoreID)
 
+        jobGraphsReachableFromRoot = {id: getJob(id) for id in reachableFromRoot}
+
```

Comment:
In practice the jobGraphs are already read into memory in the jobCache. So this does create n keys, but the values aren't copied, just referenced. So it's not all *that* bad, assuming the ids are small compared to the jobGraphs.

That said, I totally agree. The memory usage for the leader balloons if you have more than a few hundred thousand jobs to resume. We may have to make the jobCache an actual cache, and rewrite this bit so it doesn't store all jobs at once.